fighthealthinsurance/common_view_logic.py

```diff
@@ -1812,7 +1813,14 @@ async def format_response(response: dict[str, str]) -> str:
             pubmed_context=pubmed_context,
             ml_citations_context=ml_citation_context,
         )
-        filtered_appeals: Iterator[str] = filter(lambda x: x != None, appeals)
+        # Only filter for professional tone if professional_to_finish is True
+        # This ensures only provider-voice appeals are included when requested
+        if professional_to_finish:
+            filtered_appeals: Iterator[str] = filter(
+                lambda x: x is not None and is_professional_tone(x), appeals
+            )  # Filters out None and non-professional-tone appeals
+        else:
+            filtered_appeals: Iterator[str] = filter(lambda x: x is not None, appeals)  # Only filters out None
```

Comment:
I don't love this, I think we should push this logic lower if we want it so that it retries X times to try and get professional and then if it still doesn't we roll on through since this could leave with no appeals which is worse.